### PR TITLE
Move service extensions plugins samples to subdirectory

### PR DESCRIPTION
**Pull request by leonm1**

### CODE REVIEW

**Review by mpwarres — APPROVED** [review]
LGTM, though can you fix the commit message to pass the presubmit check?

File: README.md [hunk 1]
```diff
@@ -3,94 +3,6 @@
 Recipes and code samples for
 [Google Cloud Service Extensions](https://cloud.google.com/).
```
Comment (martijneken) [inline README.md:4]: Can you:
- mention the difference between callouts and plugins
- add a link to the plugins README so it's discoverable
- update the docs link to point here: https://cloud.google.com/service-extensions/docs/overview

Comment (martijneken) [reply README.md:4]: For the plugins/callouts overview, you should probably copy the text from here: https://cloud.google.com/service-extensions/docs/overview#service-extensions-types

Comment (leonm1) [reply README.md]: Done!

File: plugins/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# Google Cloud Service Extension Plugins Samples
+
+Recipes and code samples for
+[Google Cloud Service Extensions](https://cloud.google.com/).
```
Comment (msampathkumar) [inline plugins/README.md:4]:
```suggestion
[Google Cloud Service Extensions](https://cloud.google.com/service-extensions/).
```

File: README.md [hunk 3]
```diff
@@ -1,95 +1,29 @@
...
-## Rust and Cargo
-
-This project leverages
-[crate_universe](http://bazelbuild.github.io/rules_rust/crate_universe.html) to
-integrate Cargo with Bazel. In order to add new Rust library dependencies:
-
-*   Edit dependencies in Cargo.toml
-*   Regenerate Bazel targets: `$ CARGO_BAZEL_REPIN=1 bazelisk build ...`
-*   Reference libraries as `@crate_index//:<target>`
+    Cloud Load Balancing Application Load Balancers support callout extensions.
```
[the remark this answers is not in the document]
Comment (leonm1) [reply README.md:26]: 1. Done.

2. The table is a great suggestion; however, I think it'd best be added to our public docs on service extensions. I will pass the suggestion along to our docs writer!

**Pull request merged**